packages/utils/plugins-helpers/src/federation.ts

```diff
@@ -275,7 +291,10 @@ export class ApolloFederation {
  * Checks if Object Type is involved in Federation. Based on `@key` directive
  * @param node Type
  */
-function isFederationObjectType(node: ObjectTypeDefinitionNode | GraphQLObjectType, schema: GraphQLSchema): boolean {
+export function checkObjectTypeFederationDetails(
+  node: ObjectTypeDefinitionNode | GraphQLObjectType,
+  schema: GraphQLSchema
+): { resolvableKeyDirectives: readonly DirectiveNode[] } | false {
```

Comment:
I wanted to make the return type `{ resolvableKeyDirectives: readonly DirectiveNode[] } | undefined`, and make `undefined` represent the case when the node is not a object type with federation.

However, the returned value is never `undefined` (image attached) 🤔 So I've made it `false` so at least typecheck works

<img width="617" alt="Screenshot 2024-08-25 at 11 41 12 PM" src="https://github.com/user-attachments/assets/8a0dd982-8495-4be4-b6ff-a9877e5133c7">